### ISSUE TITLE
Error doing a `helm upgrade --install` from 3.1.5 to 3.2

### ISSUE THREAD

**Issue by michael-cico** [oracle/coherence-operator#488]
We merged in the chart for the COH 3.2 operator (from 3.1.5) into our [Verrazzano] repo, (https://github.com/verrazzano/verrazzano), but our upgrade scenario from 3.1.5 -> 3.2 is failing, looks like it’s complaining on the helm upgrade command about trying to change immutable selector fields.

I can repro this in our repo using  an empty kind cluster, doing the following:

* checked out the release-0.17 branch in Verrazzano (which contains the 3.1.5 chart)
* Do a helm upgrade/install of coh:
```
$ cd ../../thirdparty/charts/coherence-operator/
$ helm upgrade --install coh .
Checked out master (with 3.2) and upgraded again, and got the same result:
$ git switch master 
$ helm upgrade --install coh .
Error: UPGRADE FAILED: cannot patch "coherence-operator" with kind Deployment: Deployment.apps "coherence-operator" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"manager", "app.kubernetes.io/instance":"coherence-operator-manager", "app.kubernetes.io/name":"coherence-operator", "app.kubernetes.io/version":"3.2.0", "control-plane":"coherence"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

If I locally revert the changes around the selector labels in the 3.2 chart, the upgrade command above passes.

Using `--force` does not fix it either.  So it seems like an issue with upgrading using the COH helm chart.

**Comment by michael-cico** [comment]
I also reproduced this in the COH operator repo:

* Cloned the COH operator repo
* Checked out v3.5.1 to a branch and install it via helm:
```
$ git checkout tags/v3.1.5 -b v3.1.5
$ make helm-chart
$ cd build/_output/helm-charts/coherence-operator/
$ helm upgrade --install coh .
Release "coh" has been upgraded. Happy Helming!
NAME: coh
LAST DEPLOYED: Mon Jul 26 20:40:03 2021
NAMESPACE: default
STATUS: deployed
REVISION: 4
TEST SUITE: None
NOTES:
Coherence Operator successfully installed.
```
* Upgrade using the latest published COH operator chart
```
$ helm repo add coherence https://oracle.github.io/coherence-operator/charts
$ helm repo update
$ helm upgrade --install coh coherence/coherence-operator
Error: UPGRADE FAILED: cannot patch "coherence-operator" with kind Deployment: Deployment.apps "coherence-operator" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app.kubernetes.io/component":"manager", "app.kubernetes.io/instance":"coherence-operator-manager", "app.kubernetes.io/name":"coherence-operator", "app.kubernetes.io/version":"3.2.0", "control-plane":"coherence"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
```

**Comment by thegridman** [comment]
Fixed in v3.2.1
